iree/compiler/Dialect/HAL/Target/LLVM/LLVMAOTTarget.cpp

```diff
@@ -316,6 +345,17 @@ class LLVMAOTTargetBackend final : public TargetBackend {
           bufferAttr);
       binaryOp.mime_typeAttr(
           executableBuilder.getStringAttr("application/x-elf"));
+    } else if (!options_.staticLibraryOutput.empty()) {
```

Comment:
Please check explicitly that `staticLibraryOutput` and `linkEmbedded` are not set together, and fail if they are both set. One should not silently take precedence over the other.

I looked through https://llvm.org/docs/CommandLine.html and don't see a recommended way to do this. @benvanik any ideas? maybe...

* make the options _not_ mutually exclusive through some implementation changes
* check that both are not set during flag parsing (before running through compilation stages that would be doomed to fail)

Reply:
That works, but I'd rather it get checked during flag parsing (so we don't spent seconds-minutes compiling just to fail because flags were misconfigured).

Reply:
Makes sense